src/mlpack/core/math/random.hpp

```diff
@@ -66,6 +66,31 @@ inline double Random(const double lo, const double hi)
 }
 
 /**
+ * Generates a binomial random number in the specified range.
+ */
+inline double BinomialRandom(const int input)
+{
+  if(Random() > input)
+    return 1;
+  else
+    return 0;
+}
+
+/**
+ * Generates a binomial random number in the specified range.
+ */
+template<typename InputDatatype, typename OutputDatatype>
```

Comment:
I agree, you can just use ``for_each``or ``imbue`` to do the same without introducing another function.